similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java

```diff
@@ -177,6 +194,39 @@ public int compare(PatientSimilarityView o1, PatientSimilarityView o2)
         return results;
     }
 
+    private boolean filterPatient(Patient matchPatient, Family family, EntityReference refOwner,
+        String requiredConsentId)
+    {
+        if (matchPatient == null) {
+            // Leftover patient in the index, should be removed
+            return true;
+        }
+        // filter out patients from the same family
+        if (family != null && family.isMember(matchPatient)) {
```

Comment:
This does not work correctly for remote patients. I just checked how `isMember()` works (see https://github.com/phenotips/phenotips/blob/master/components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/PhenotipsFamily.java#L140) - it just compares patient IDs. Remote patients may have an ID that matches an ID of some family member, but it still does not mean that patient is in a local family. Need to add a check first that the patient is local

Reply:
@allasm `matchedPatient` can't possibly be remote. For reference remote patients family is null.

Reply:
hmm, right, my bad. I will remove this comment then